### ISSUE TITLE
Security: create up-to-date tagged release

### ISSUE THREAD

**Issue by lucasgonze** [benoitc/gunicorn#2828]
There has not been a tagged release since Apr 28, 2021. As a result most users of this package cannot access the security fix in https://github.com/benoitc/gunicorn/pull/2581. 

There has been a lot of work and many merges since then, but that is out of scope for me personally and for this ticket.

**Comment by bt** [comment]
@benoitc - Is this something you'd be able to comment on? Is there something that needs to happen before we can tag a new release?

**Comment by lucasgonze** [comment]
@benoitc Do you need assistance from contributors? Can we help lighten your load?

**Comment by naktinis** [comment]
@benoitc is there something preventing a new release? If you are currently busy with other things, maybe contributors could help?

**Comment by lucasgonze** [comment]
Pinged on Twitter: https://twitter.com/lucas_gonze/status/1563217351608115200

**Comment by mball-agathos** [comment]
@benoitc If a full-release is too burdensome, it would be highly useful to have a point-release (e.g., `20.1.1`) containing just a cherry-pick of https://github.com/benoitc/gunicorn/pull/2581 (and any other high-priority fixes).  I could help by creating a pull-request with the cherry-pick, but we'd need you or another maintainer to generate the release from this update.  Would this be easier than doing another major release?  Thanks for your help in maintaining this very useful open-source package!

**Comment by riptusk331** [comment]
Any traction on this??

**Comment by binaryDiv** [comment]
@benoitc I really understand that maintaining an open-source project is a lot of voluntary work, and I appreciate all the work that went into this project.

However, this is a project with a lot of contributors, and it is widely used in production. It cannot take a full year (or even longer) to just make a patch release with very critical fixes (that are already done and merged, just not released).

If you don't have the time or energy (or simply don't want) to maintain this repository, this is absolutely fine and understandable. But in that case there should be at least a co-maintainer who can accept pull requests and release new versions.

From the comments above it appears like there are multiple people who would love to help with that.

**Comment by fabswt** [comment]
I spent hours in dependency hell, so here's a quick recap for anyone new to the 'party' who needs gunicorn+eventlet+dnspython:

Got things working with this:

```
gunicorn @ git+https://github.com/benoitc/gunicorn.git@792edf6
eventlet==0.33.3
dnspython==2.3.0
```

I hope this'll save someone some time.

[Detailed notes here](https://www.dropbox.com/s/euv3884t4b7gykf/EVENTLET-GUNICORN-DNSPYTHON%20HELL.md?dl=0)

**Comment by NeilujD** [comment]
Thanks for the hint @fabswt !
Which version of Python are you using ?

**Comment by fabswt** [comment]
@NeilujD I was on Python 3.9 but, with eventlet and dnspython now up-to-date, will be able to switch back to 3.10. I realize the "detailed notes" link broke after I moved the file, I just fixed it – it includes details about the Python versions.

**Comment by lars1264** [comment]
Thanks @fabswt,
works for me. Installed it via 
pip install -I gunicorn git+https://github.com/benoitc/gunicorn.git@792edf6

**Comment by bt** [comment]
Hi @benoitc, this issue is open.

**Comment by benoitc** [comment]
@bt f you want to help, please test master and report any issue you still have or comment current opened issue. This will be much appreciated.

**Comment by benoitc** [comment]
I am closing this issue since a release is about to land.  Master already include it the needed change. 

Also while I understand the frustration,  this is not the proper way to handle "security" issues. Please drop a mail to security@ next time. Or directly to me if you want to.

**Comment by fabswt** [comment]
I'm confused, why is this closed?

- The README still says to install with `pip install gunicorn` but PyPi is still offering 20.1.0 (from Apr 28, 2021) which does not include the security fix mentioned in the OP and just won't work with recent versions of eventlet.
-AFAIK using a hash/patch (`gunicorn @ git+https://github.com/benoitc/gunicorn.git@792edf6`) is still the only way to get gunicorn to run.

Closing this only adds to the confusion.

**Comment by devopstales** [comment]
@benoitc Why this issue is closed? There is no new release since 20.1.0 (from Apr 28, 2021) not on PyPi nor on the repo.

**Comment by lanmaster53** [comment]
> I am closing this issue since a release is about to land.

From May 7th. Any updates @benoitc?

**Comment by binaryDiv** [comment]
> I am closing this issue since a release is about to land.

This issue is about creating a new release. Issues are supposed to be closed *after* they're done, not before that. I really don't understand why this issue was closed.

Also, I'm sorry, but I think it's time to either fork this project or just switch to a different WSGI server implementation. Our company is currently evaluating alternatives, because the state of maintenance of this project just isn't suitable for production environments.
